lib/detectProjectRoot.js

```diff
@@ -5,6 +5,16 @@ var path = require("path");
 
 module.exports = function(cwd) {
   var projectRootDir = cwd || process.cwd();
+  if (typeof projectRootDir !== "string") {
+    throw new Error(
+      "Project Root Directory is not a string. Please change your directory under firebase.json to be a valid string for the source paramter."
+    );
+  }
+  if (!projectRootDir) {
```

Comment:
The second if statement is not necessary, since `typeof undefined` would return "undefined" and therefore be caught by the first if statement. Could you also revise the error message to:

```
      throw new FirebaseError(
        chalk.bold("firebase.json") +
          " is missing a " +
          chalk.bold("source") +
          " field in the " +
          chalk.bold("functions") +
          " object. Please add one, and specify the source directory for your functions. For example:\n" +
          chalk.bold('\t"functions": {\n\t\t"source": "functions"\n\t}')
      );
```

You will also need to add this at the top of the file:
`var FirebaseError = require("../../error");`